### DISCUSSION TITLE
Cannot enable Github Enterprise OAuth

### DISCUSSION THREAD

**Discussion by ELatestark** [WeblateOrg/weblate#12327] [Q&A] [answered]
Hey! Couple hours ago I was able to set Github OAuth and thought it won't be much difference, but I'm completely lost with GHE auth.
Button "Sign in with: Github Enterprise" does literally nothing and then I took a look at network traffic.
![Снимок экрана от 2024-08-23 23-16-28](https://github.com/user-attachments/assets/11f22018-8b2a-4196-9091-5edac186ea38)
I noticed location field in reponse, just copy-pasted it and then my browser was able to proceed to auth webpage, but then:
![Снимок экрана от 2024-08-23 23-20-12](https://github.com/user-attachments/assets/84b8c470-f878-4228-bd98-349c221fcdf9)

I'm also unsure about my docker variables:
![Снимок экрана от 2024-08-23 23-21-26](https://github.com/user-attachments/assets/321708eb-ce2e-42c3-bfc5-53a5af12c6d3)
What's the purpose of SOCIAL_AUTH_GITHUB_ENTERPRISE_SCOPE = ["user:email"] ? Is it expected to fill "user:email" with actual user and email info? I was able to make OAuth work with Github without such variable and was expecting the same from GHE.

According to documentation, there are [different variables for Github for Teams and Github for Organizations. ](https://docs.weblate.org/en/latest/admin/auth.html#github-authentication), but what if I'm in Github Enterprise for Organizations? [There's a link to Python Social auth documentation](https://python-social-auth.readthedocs.io/en/latest/backends/github_enterprise.html) and it states that there are different variables for GHE for Organizations with SOCIAL_AUTH_GITHUB_ENTERPRISE_ORG_* prefix, but these variables weren't stated in Weblate docs.

Upd.: I was able to login by copying location field in reponse, seems like the SOCIAL_AUTH_GITHUB_ENTERPRISE_SCOPE var was set in a wrong way. I got redirected at auth page and logged in with my GHE account. The only question left: why it's not redirecting me automatically?

Upd.: continuing my investigation I bumped into this error when I'm pressing login button:

> Refused to send form data to 'http://***.***.***.***/accounts/login/github-enterprise/' because it violates the following Content Security Policy directive: "form-action 'self' github.com".

![Снимок экрана от 2024-08-24 01-07-14](https://github.com/user-attachments/assets/e6db173d-be90-4006-a66b-fecd01a0e003)

Then I found WEBLATE_CSP_* variables in docs. I guess it looks related to my problem. But I don't see a variable related to form-action field. What do I do?

**Comment by avm99963** [comment, marked as answer]
As for the button not doing anything, it was reported in bug #12321, which has been fixed by nijel in https://github.com/WeblateOrg/weblate/commit/45e558cdfa2f7f6b64616e544fdb05367fe10121 (as of now unreleased).

**Comment by nijel** [comment]
Weblate has a GitHub-specific code to fetch all user e-mail addresses from GitHub, something similar will be probably needed for GitHub Enterprise:

https://github.com/WeblateOrg/weblate/blob/768cd3b26cf50c05878ce23c53c20436a7ec7300/weblate/accounts/pipeline.py#L114-L118

**Comment by ELatestark** [comment]
Fixed in 5.7.1
